collector/test/NetworkStatusNotifierTest.cpp

```diff
@@ -156,14 +251,114 @@ TEST(NetworkStatusNotifier, SimpleStartStop) {
 
   net_status_notifier->Start();
 
-  // we could send probe events using conn_tracker->UpdateConnection
-
   EXPECT_TRUE(sem.try_acquire_for(std::chrono::seconds(5)));
 
-  running = false;
   net_status_notifier->Stop();
+}
 
-  SUCCEED();
+/* This test checks whether deltas are computed appropriately in case the "known network" list is received after a connection
+   is already reported (and matches one of the networks).
+   - scrapper initialy reports a connection
+   - the connection is reported
+   - we feed a "known network" into the NetworkStatusNotifier
+   - we check that the former declared connection is deleted and redeclared as part of this network */
+TEST(NetworkStatusNotifier, UpdateIPnoAfterglow) {
+  bool running = true;
+  MockCollectorConfig config;
+  std::shared_ptr<MockConnScraper> conn_scraper = std::make_shared<MockConnScraper>();
+  auto conn_tracker = std::make_shared<ConnectionTracker>();
+  auto comm = std::make_shared<MockNetworkConnectionInfoServiceComm>();
+  std::function<void(const sensor::NetworkFlowsControlMessage*)> network_flows_callback;
+  Semaphore sem(0);  // to wait for the service to accomplish its job.
+
+  // the connection as scrapped (public)
+  Connection conn1("containerId", Endpoint(Address(10, 0, 1, 32), 1024), Endpoint(Address(139, 45, 27, 4), 999), L4Proto::TCP, true);
+  // the same server connection normalized
+  Connection conn2("containerId", Endpoint(Address(), 1024), Endpoint(Address(255, 255, 255, 255), 0), L4Proto::TCP, true);
+  // the same server connection normalized and grouped in a known subnet
+  Connection conn3("containerId", Endpoint(Address(), 1024), Endpoint(IPNet(Address(139, 45, 0, 0), 16), 0), L4Proto::TCP, true);
+
+  UnorderedMap<Address::Family, std::vector<IPNet>> known_networks = {{Address::Family::IPV4, {IPNet(Address(139, 45, 0, 0), 16)}}};
```

Comment:
Maybe I'm missing something, where is this variable being used?

Reply:
You are right, it is not !

I prepared this "known network" variable in order to inject it during the test and re-made it here: https://github.com/stackrox/collector/blob/olivier/ROX-11097-test-pubIP-change-in-NetworkStatusNotifier/collector/test/NetworkStatusNotifierTest.cpp#L327

I will remove this definition.